launch/launch/launcher.py

```diff
@@ -225,7 +254,18 @@ def _run(self):
                         except ProcessLookupError:
                             pass
 
-            yield from asyncio.wait(all_futures.keys())
+            kwargs = {}
+            if not isinstance(threading.current_thread(), threading._MainThread):
+                # wake up periodically if we are not in the main thread
```

Comment:
I looked over this again and I don't understand the question. This same check was being used in this function already. Maybe you could describe a scenario where this wouldn't work?